### ISSUE TITLE
ECS containers incompatible with the latest Amazon Linux

### ISSUE THREAD

**Issue by cs1jmc** [aws-ia/cfn-ps-duo-mfa#13]
https://github.com/aws-ia/cfn-ps-duo-mfa/blob/96b65e1b6f81d98116f87c11b4a672c952941bc6/scripts/source/docker/Dockerfile#L1

When deploying from the latest AL image 2023 has become the default causing an OpenSSL error that bricks the tasks:

```bash
An exception prevented the connectivity tool from running: Unknown OpenSSL error. This error is commonly encountered when another library is not cleaning up the OpenSSL error stack. If you are using cryptography with another library that uses OpenSSL try disabling it before reporting a bug. Otherwise please file an issue at https://github.com/pyca/cryptography/issues with information on how to reproduce this. ([<OpenSSLError(code=126615813, lib=15, reason=786693, reason_text=init fail)>])
```

I'm not sure what changes need to be made for the deployment to work with the 2023 AmazonLinux builds but as a workaround for now specifying AL2 works `FROM public.ecr.aws/amazonlinux/amazonlinux:2`

**Comment by kyvidmar** [comment]
Thanks @cs1jmc for anyone using the e2e cloudformation template with ECS I was able to repair Duo with the following steps:

1. Navigate AWS Services Search
2. CodeCommit Service
3. Duo-AuthProxy
4. Navigate Docker
5. Navigate Dockerfile
6. Edit line above to  `FROM public.ecr.aws/amazonlinux/amazonlinux:2` and save
7. Navigate Amazon EventBridge Service
8. Find Job that builds container
9. Edit Event Schedule Cron so that it executes
10. Wait for image to finish building and deploying
11. Restart ECS tasks

AWS Services -> CodeCommit -> Duo-AuthProxy -> Docker -> Dockerfile -> Edit to amazonlinux:2

**Comment by vsnyc** [comment]
Fixed in #14
